flutter/lib/firebase/rest_helper.dart

```diff
@@ -0,0 +1,29 @@
+import 'dart:convert';
+import 'dart:io' show HttpStatus;
+
+import 'package:firebase_dart/firebase_dart.dart';
+import 'package:http/http.dart' as http;
+
+import 'package:mlperfbench/data/extended_result.dart';
+import 'config.gen.dart';
+
+class RestHelper {
+  final UserCredential userInfo;
+
+  RestHelper(this.userInfo);
+
+  Future<void> upload(ExtendedResult result) async {
+    const path = '/v0/upload';
+    final jsonResult = result.toJson();
+    var response =
+        await http.post(Uri.parse('${FirebaseConfig.functionsUrl}$path'),
+            headers: {
+              'Authorization': '${await userInfo.user!.getIdToken()}',
```

Comment:
Force unwrap of user may cause crash.

Reply:
Fixed, I added a separate check.
Though, this function is expected to throw errors, so calls to it were wrapped into try-catch block from the start. Explicit check just makes the error more readable.

Reply:
I did not test it but I think force unwrap will crash the app and user has no idea what happened, but with try-catch we can show a popup with an error message.